### DISCUSSION TITLE
Private packages in go.mod using whitsource github renovate

### DISCUSSION THREAD

**Discussion by alexcurtin** [renovatebot/renovate#10946] [Request Help]
If a projects go.mod has private repository dependencies and we're using the whitesource renovate bot installed via GIthub Apps, how do I enable the bot to checkout / verify these dependencies? We're currently getting:

```
Command failed: docker run --rm --name=renovate_go --label=renovate_child -v "/mnt/renovate/gh/redacted":"/mnt/renovate/gh/redacted" -v "/tmp/renovate-cache":"/tmp/renovate-cache" -v "/tmp/renovate-cache/others/go":"/tmp/renovate-cache/others/go" -e GOPATH -e GOFLAGS -e CGO_ENABLED -w "/mnt/renovate/gh/redacted" docker.io/renovate/go:1.16.6 bash -l -c "go get -d ./... && go install github.com/marwan-at-work/mod/cmd/mod@latest && mod upgrade --mod-name=github.com/go-redis/redis -t=7 && go mod tidy && go mod tidy"
go: github.com/redacted@v1.0.0: reading github.com/redacted/go.mod at revision v1.0.0: unknown revision v1.0.0
````

**Comment by rarkins** [comment]
It should work if:
- They're both in the same github org
- They're both private
- They both have Renovate installed

Is that the case?

**Comment by jsol** [comment]
I am seeing the same thing. We use a mono-repo (on github), and when renovate tries to update an internal lib (in the same repo) we see the same issue.

I speculate that go has started to check public sources of cache/sums in more cases as I had to modify our builds with GOPRIVATE envs for the build step itself (after a go mod download), but that might be unrelated.

**Comment by alexcurtin** [comment]
Hi @rarkins 

Yes both repositories/dependencies are in the same github org, they're both private, and they both have renovate installed. Still seeing the same issues.

**Reply by rarkins** [reply]
There previously would be a git insteadOf command run but I don't see it here. Need to dig into that

**Comment by rarkins** [comment]
@shegox @viceice I think it's possible that we broke things in #10750

The new docs state, `For each comma-separated module path in `GOPRIVATE` or `golang.registryUrls` Renovate will look for credentials in `hostRules`.`

But previously it "just worked" because Renovate would always write the app token using git insteadof.

**Reply by Shegox** [reply]
Yes, I think the problem is that when setting `GOPRIVATE` for all of github.com all modules would be fetched directly from source instead of the proxy which is most likely undesired. Therefore `registryUrls` is used to determine which modules to fetch directly from the source. I think it might make sense to automatically add `github.com/$ORG` to the list of the `registryUrls`, thus letting renovate fetch these dependencies from source.

Or, but I haven't verified that, it would be enough to have the authentication available (e.g. how it was previously) to just add for github.com always credentials if possible and the go command is clever enough to default to the source if the goproxy doesn't have the the mod available. On GitHub Enterprise Server that didn't work, but might work on github.com.

**Comment by rarkins** [comment]
Regression error was found, and now rolled back. The app will be updated later today
